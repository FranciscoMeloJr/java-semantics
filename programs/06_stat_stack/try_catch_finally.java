// Testing try with  both catch and finally clause

public class try_catch_finally {

  public static void main(String[] args) {
    try {
      try {
        System.out.println("no exception first time");
      } catch (RuntimeException e3) {
        System.out.println("caught exception: "+e3.toString());
      } finally {
        System.out.println("finally after no exception");
      }

      if(true) throw new RuntimeException("");

      System.out.println("No exception second time");
    } catch (RuntimeException e3) {
      System.out.println("caught exception: " + e3.toString());
    } finally {
      System.out.println("finally after exception");
    }
    System.out.println("Done!");
  }
}

// no exception first time
// finally after no exception
// caught exception: java.lang.RuntimeException:
// finally after exception
// Done!
