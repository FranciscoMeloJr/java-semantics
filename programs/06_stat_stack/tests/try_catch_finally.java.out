no exception first time
finally after no exception
caught exception: java.lang.RuntimeException: 
finally after exception
Done!
